php/classes/DB.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+require_once __DIR__ . '/../inc/configDB.php';
+
+class DB {
+  // (A) CONSTRUCTOR - CONNECT TO DATABASE
+  private $pdo = null;
+  private $stmt = null;
+  public $msg = "";
+  function __construct () {
+    try {
+      $this->pdo = new PDO(
+        "mysql:host=".DB_HOST.";dbname=".DB_NAME.";charset=".DB_CHARSET, 
+        DB_USER, DB_PASSWORD, [
+          PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION,
+          PDO::ATTR_DEFAULT_FETCH_MODE => PDO::FETCH_ASSOC
+        ]
+      );
+    } catch (Exception $ex) { die($ex->getMessage()); }
+  }
+
+  // (B) DESTRUCTOR - CLOSE DATABASE CONNECTION
+  function __destruct () {
+    if ($this->stmt!==null) { $this->stmt = null; }
+    if ($this->pdo!==null) { $this->pdo = null; }
+  }
+
+  function queryFetchAll ($sql) {
+    return $this->pdo->query($sql)->fetchAll(PDO::FETCH_ASSOC);
+  }
+
+  function queryFetch ($sql) {
+    return $this->pdo->query($sql)->fetch();
+  }
+
+  function getFirstReceiptToCheck() {
+    $sql = "SELECT * 
+    FROM receipts
+    WHERE checked=false
+    ORDER BY `date_emission` ASC, `id` ASC
+    LIMIT 1;";
+      
+    return $this->pdo->query($sql)->fetch(PDO::FETCH_ASSOC);
+  }
+
+  function saveReceipt() {
+    try {
+        $_POST['ischecked'] === "true" ? $isChecked = 1 : $isChecked = 0;
+        if (isset($_POST['checkReceiptAndSelectNext'])) {
+            $isChecked = 1;
+        }
+
+        if ($_POST['receiptid'] == "") { // new receipt
+            $sth = $this->pdo->prepare('INSERT INTO `receipts`(`photo_name`, `photo_data`, `date_emission`, `category`, `provider`, `montant_ttc`, `tva`, `checked`, `description`) VALUES (:photo_name, :photo_data, :date, :receiptCategory, :provider, :amountTTC, :tva, :isChecked, :description)');
+            
+            $photoData = file_get_contents($_FILES['photo']['tmp_name']);
```

Comment:
there seem to be code duplication inside this function. Be careful not to repeat yourself (checkout the [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) principle)